crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -333,13 +344,9 @@ impl InstancePool {
             memories: MemoryPool::new(module_limits, instance_limits, tunables)?,
             tables: TablePool::new(module_limits, instance_limits)?,
             empty_module: Arc::new(Module::default()),
+            module_limits: module_limits.clone(),
```

Comment:
I think we can actually avoid saving this internally since initialization below should I think take into account the specific module being instantiated. For example there's a call to `PrimaryMap::with_capacity(self.module_limits.memories as usize)` but that's not necessary if the module being instantiated has fewer memories than the limits.

I think this could probably be achieved by extracting [this logic](https://github.com/bytecodealliance/wasmtime/blob/90bfa123e095371691c43679ae7c8f11380d8bf6/crates/runtime/src/instance/allocator.rs#L693-L711) to a helpe method and using that instead of the `initialize` method below perhaps?

Reply:
Done -- extracted a common core between this and the on-demand allocator to build a raw `Instance`.